### PR TITLE
Increase experimental `messageArgs` support

### PR DESCRIPTION
**Pull request by chloerice**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/6453,
Closes https://github.com/stylelint/stylelint/issues/6454,
Closes https://github.com/stylelint/stylelint/issues/6455,
Closes https://github.com/stylelint/stylelint/issues/6461, and
Closes https://github.com/stylelint/stylelint/issues/6462

> Is there anything in the PR that needs further explanation?

No, it's modeled after the `color-no-hex` changes in #6312

### CODE REVIEW

**Comment by ybiquitous** [comment]
@chloerice Thanks for creating the PR. I believe PR #6464 should fix the CI failures.

File: .changeset/several-cucumbers-wonder.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Added: `declaration-property-value-disallowed-list` `messageArgs` support
+Added: `declaration-property-unit-disallowed-list` `messageArgs` support
+Added: `function-disallowed-list` `messageArgs` support
+Added: `at-rule-disallowed-list` `messageArgs` support
+Added: `property-disallowed-list` `messageArgs` support
```
Comment (ybiquitous) [inline .changeset/several-cucumbers-wonder.md:5]: [suggest] Thanks for adding the entry. Could you please update it as follows to work `changeset` later?

```suggestion
---
"stylelint": minor
---

Added: custom message formatting for `at-rule-disallowed-list`, `declaration-property-unit-disallowed-list`, `declaration-property-value-disallowed-list`, `function-disallowed-list`, and `property-disallowed-list`
```

File: lib/rules/at-rule-disallowed-list/README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Specify a list of disallowed at-rules.
  * At-rules like this */
 ```
 
-## Options
+The [`message` secondary option](../../../docs/user-guide/configure.md#message) can accept the arguments of this rule.## Options
```
Comment (ybiquitous) [inline lib/rules/at-rule-disallowed-list/README.md:12]: It looks like a typo.

```suggestion
The [`message` secondary option](../../../docs/user-guide/configure.md#message) can accept the arguments of this rule.

## Options
```

**Review by ybiquitous — CHANGES_REQUESTED** [review]
I've merged #6464. Please merge the main branch into this PR branch. CI should work.

**Review by ybiquitous — APPROVED** [review]
Thank you. LGTM 👍🏼

**Comment by chloerice** [comment]
Thanks for all your help @ybiquitous!! 🙏🏽

**Pull request merged**